homeassistant/components/tplink/light.py

```diff
@@ -72,192 +75,318 @@ def brightness_from_percentage(percent):
     return (percent * 255.0) / 100.0
 
 
+LightState = NamedTuple(
+    "LightState",
+    (
+        ("state", bool),
+        ("brightness", int),
+        ("color_temp", float),
+        ("hs", Tuple[int, int]),
+        ("emeter_params", dict),
+    ),
+)
+
+
+LightFeatures = NamedTuple(
+    "LightFeatures",
+    (
+        ("sysinfo", Dict[str, Any]),
+        ("mac", str),
+        ("alias", str),
+        ("model", str),
+        ("supported_features", int),
+        ("min_mireds", float),
+        ("max_mireds", float),
+    ),
+)
+
+
 class TPLinkSmartBulb(Light):
     """Representation of a TPLink Smart Bulb."""
 
+    _light_state: LightState
```

Comment:
Why put the typing here? We can type the instance attributes directly in init method.